lib/media/segment_index.js

```diff
@@ -694,6 +694,18 @@ shaka.media.MetaSegmentIndex = class extends shaka.media.SegmentIndex {
     this.indexes_ = [];
   }
 
+  /**
+   * @override
+   * @export
+   */
+  forEachTopLevelReference(fn) {
+    for (const index of this.indexes_) {
+      for (const reference of index.references) {
+        fn(reference);
+      }
+    }
```

Comment:
I'd suggest this to make it work with possible different `SegmentIndex` implementations:
```suggestion
    for (const index of this.indexes_) {
      index.forEachTopLevelReference(fn);
    }
```

Reply:
Done!